service/authorization/authorization.go

```diff
@@ -359,40 +360,107 @@ func (as *AuthorizationService) GetDecisions(ctx context.Context, req *authoriza
 	return rsp, nil
 }
 
-func (as *AuthorizationService) GetEntitlements(ctx context.Context, req *authorization.GetEntitlementsRequest) (*authorization.GetEntitlementsResponse, error) {
-	as.logger.DebugContext(ctx, "getting entitlements")
-	request := attr.GetAttributeValuesByFqnsRequest{
-		WithValue: &policy.AttributeValueSelector{
-			WithSubjectMaps: true,
-		},
+// makeSubMapsByValLookup creates a lookup map of subject mappings by attribute value ID.
+func makeSubMapsByValLookup(subjectMappings []*policy.SubjectMapping) map[string][]*policy.SubjectMapping {
+	// map keys will be attribute value IDs
+	lookup := make(map[string][]*policy.SubjectMapping)
+	for _, sm := range subjectMappings {
+		val := sm.GetAttributeValue()
+		id := val.GetId()
+		// if attribute value ID exists
+		if id != "" {
+			// append the subject mapping to the slice of subject mappings for the attribute value ID
+			lookup[id] = append(lookup[id], sm)
+		}
```

Comment:
is it an error case if `id == ""`, which looks like it would silently fail here, or is that intentional?

Reply:
The id check also checks that the attribute value exists because of the function chaining. We could add a warning here in the future like `missing attribute value on subject mapping - db is potentially in a bad state`. I don't think it should necessarily error out the function since it's in a loop.